### PR TITLE
Cut/Copy line if at left margin.

### PR DESCRIPTION
**Pull request by dudeNumber4**

### CODE REVIEW

**Comment by alefragnani** [comment]
Hi @dudeNumber4 ,

First of all, thanks for your PR. Please, open an issue about the feature you are proposing, so we can gather feedback from other users.

About this specific feature, it should need at least an _opt-in_ setting. Simply because I created the extension exactly to avoid copying the entire line when nothing is selected 😆 .

Thank you

**Review by alefragnani — CHANGES_REQUESTED** [review]

File: src/commands.ts [hunk 1]
```diff
@@ -37,8 +40,19 @@ export function registerCommands() {
         if (!canExecuteOperation(Operations.Cut)) { return; }
 
         const editor = window.activeTextEditor!;
-        if (editor.selection.isEmpty) {			
-            if (selectWordAtCursorPosition(editor)) {
+        if (editor.selection.isEmpty) {	
+            if (isAtStartOfLine(editor)) {
```
[the remark this answers is not in the document]
Comment (dudeNumber4) [reply src/commands.ts:44]: The refactor doesn't work.  If start of line and configured, we finish the entire command, not just select.  So you'd have to check conditions yet again after calling the new func.

File: src/commands.ts [hunk 2]
```diff
@@ -18,14 +18,17 @@ export function registerCommands() {
...
+    commands.registerCommand("copy-word.copy", async () => {
 
         if (!canExecuteOperation(Operations.Copy)) { return; }
 
         const editor = window.activeTextEditor!;
         if (editor.selection.isEmpty) {
-            if (selectWordAtCursorPosition(editor)) {
-                env.clipboard.writeText(editor.document.getText(editor.selection));
+            if (isAtStartOfLine(editor)) {
+                await env.clipboard.writeText(getCurrentLine(editor));
```
[the remark this answers is not in the document]
Comment (dudeNumber4) [reply src/commands.ts]: Amazing that I read right over `editor.action.clipboardPasteAction`.  I was just thinking, "why doesn't VS Code work as expected?

**Comment by dudeNumber4** [comment]
@alefragnani , can this be merged?  Issues have been addressed.

**Comment by alefragnani** [comment]
Sorry for the late reply. Had some other priorities to deal with.
I'm planning to look the PR this weekend.

Thank you

**Pull request merged**